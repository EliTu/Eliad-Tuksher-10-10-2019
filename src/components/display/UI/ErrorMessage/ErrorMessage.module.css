@import '../../../../variables.css';

.ErrorMessageStyles {
    display: block;
    width: 50%;
    padding: 1rem 2rem;
    color: var(--color-main-black);
}

.ErrorMessageStyles p:nth-child(1) {
    font-size: var(--text-size-extra-large);
    font-weight: 800;
}

.ErrorMessageStyles p:nth-child(2) {
    font-size: var(--text-size-large);
    font-size: 600;
}

.ErrorMessageStyles p:nth-child(3) {
    font-size: var(--text-size-medium);
}

.ErrorMessageStyles p:nth-child(4) {
    font-size: var(--text-size-medium);
}

.ErrorMessageStyles span {
    color: rgba(248, 129, 129, 0.822);
    font-weight: 700;
}