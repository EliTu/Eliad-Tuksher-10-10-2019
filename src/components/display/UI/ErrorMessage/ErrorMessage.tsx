import React from 'react';

import styles from './ErrorMessage.module.css';

interface IProps {
    errorDetails: string;
	errorType: string;
}

export const ErrorMessage: React.FC<IProps> = ({ errorDetails, errorType }) => {
	const { ErrorMessageStyles } = styles;

	let headMessage;
	let secondaryMessage;
	switch (errorType) {
		case 'searchError':
			headMessage = 'Oops...';
			secondaryMessage = 'Could not perfrom search!';
			break;
		case 'currentWeatherError':
		case 'fiveDaysForecastError':
			headMessage = 'Oops...';
			secondaryMessage = 'Something went wrong with the request :(';
			break;
		default:
			headMessage = '';
			secondaryMessage = '';
			break;
	}

	return (
		<div className={ErrorMessageStyles}>
			<p>{headMessage}</p>
			<p>{secondaryMessage}</p>
			<p>
				The reason seems to be: <span>{errorDetails}</span>
			</p>
			<p>Please try again in a short while!</p>
		</div>
	);
};

export default ErrorMessage;
