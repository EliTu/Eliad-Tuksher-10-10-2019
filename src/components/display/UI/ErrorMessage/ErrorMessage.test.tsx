import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import initialShallowRender from '../../../../utilities/test-utilities/initialShallowRender';
import ErrorMessage from './ErrorMessage';

describe('ErrorMessage component', () => {
	let component: ShallowWrapper;
	beforeEach(() => (component = shallow(<ErrorMessage errorDetails={''} />)));

	it('should render without errors', () => {
		initialShallowRender(component, 'ErrorMessageStyles');
	});

	it('should render 4 p tags', () => {
		expect(component.children().length).toBe(4);
	});

	it('should render a search error message if errorType is searchError', () => {
		component.setProps({
			errorDetails: 'network error',
		});

		expect(component).toMatchSnapshot();
		expect(
			component
				.children()
				.at(0)
				.text()
		).toBe('Oops...');
		expect(
			component
				.children()
				.at(1)
				.text()
		).toBe('Something went wrong with the server request :(');
		expect(
			component
				.children()
				.at(2)
				.text()
		).toBe('The reason seems to be: network error');
		expect(
			component
				.children()
				.at(3)
				.text()
		).toBe('Please try again in a short while!');
	});

	it('should render a currentWeather error message if errorType is currentWeatherError', () => {
		component.setProps({
			errorDetails: 'internal error',
		});

		expect(component).toMatchSnapshot();
		expect(
			component
				.children()
				.at(0)
				.text()
		).toBe('Oops...');
		expect(
			component
				.children()
				.at(1)
				.text()
		).toBe('Something went wrong with the server request :(');
		expect(
			component
				.children()
				.at(2)
				.text()
		).toBe('The reason seems to be: internal error');
		expect(
			component
				.children()
				.at(3)
				.text()
		).toBe('Please try again in a short while!');
	});

	it('should render a fiveDaysForecast error message if errorType is fiveDaysForecastError', () => {
		component.setProps({
			errorDetails: 'source error',
		});

		expect(component).toMatchSnapshot();
		expect(
			component
				.children()
				.at(0)
				.text()
		).toBe('Oops...');
		expect(
			component
				.children()
				.at(1)
				.text()
		).toBe('Something went wrong with the server request :(');
		expect(
			component
				.children()
				.at(2)
				.text()
		).toBe('The reason seems to be: source error');
		expect(
			component
				.children()
				.at(3)
				.text()
		).toBe('Please try again in a short while!');
	});
});
