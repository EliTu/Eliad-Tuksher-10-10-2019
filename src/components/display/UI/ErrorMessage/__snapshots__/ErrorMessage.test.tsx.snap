// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component should render a currentWeather error message if errorType is currentWeatherError 1`] = `
<div
  className="ErrorMessageStyles"
>
  <p>
    Oops...
  </p>
  <p>
    Something went wrong with the request :(
  </p>
  <p>
    The reason seems to be: 
    <span>
      internal error
    </span>
  </p>
  <p>
    Please try again in a short while!
  </p>
</div>
`;

exports[`ErrorMessage component should render a fiveDaysForecast error message if errorType is fiveDaysForecastError 1`] = `
<div
  className="ErrorMessageStyles"
>
  <p>
    Oops...
  </p>
  <p>
    Something went wrong with the request :(
  </p>
  <p>
    The reason seems to be: 
    <span>
      source error
    </span>
  </p>
  <p>
    Please try again in a short while!
  </p>
</div>
`;

exports[`ErrorMessage component should render a search error message if errorType is searchError 1`] = `
<div
  className="ErrorMessageStyles"
>
  <p>
    Oops...
  </p>
  <p>
    Could not perfrom search!
  </p>
  <p>
    The reason seems to be: 
    <span>
      network error
    </span>
  </p>
  <p>
    Please try again in a short while!
  </p>
</div>
`;

exports[`ErrorMessage component should render without errors 1`] = `
<div
  className="ErrorMessageStyles"
>
  <p />
  <p />
  <p>
    The reason seems to be: 
    <span />
  </p>
  <p>
    Please try again in a short while!
  </p>
</div>
`;
